src/data-table/data-table.tsx

```diff
@@ -124,10 +131,11 @@ export const DataTable = ({
             <HeaderCell
                 data-testid={getDataTestId("header-selection")}
                 $clickable={false}
+                $isCheckbox={true}
             >
                 <CheckBoxWrapper>
                     {enableSelectAll && (
-                        <CheckboxElement
+                        <Checkbox
```

Comment:
there should be an indeterminate state set for the case when not all items are selected

Reply:
I have updated indeterminate changes with next changes, Can I do the update of the indeterminate in there?